### ISSUE TITLE
Windows: No warning about unsaved changes

### ISSUE THREAD

**Issue by Geweldig** [onivim/oni2#610]
On Windows 10, if I have a file open in which I have made changes I am not warned about this before closing the program. This means all changes are lost, without having that behaviour be confirmed by the user.

**Comment by bryphe** [comment]
This is a really embarassing bug. At the very least - we should show our own dialog listing which files are 'unsaved' and give the user the opportunity to save them.

The work here would be:
- Create a `<Dialog />` UI component that we could use to handle this.
- Hook into the lifecycle events to prevent closing (Revery has a `Window.setCanQuitCallback` that lets us hook into this, so we can prevent the app from closing based on a condition: https://github.com/revery-ui/revery/blob/7558f7c4055ce3a1f42fb1c63c445af6a90996f4/src/Core/Window.re#L621)

So basically - we could hook into that `Revery.Window.setCanQuitCallback`, do a check for any unsaved files - if there are no unsaved files, we return `true` and let the app quit. If there are unsaved files, we return `false` and dispatch an appropriate action.

**Comment by glennsl** [comment]
How about using the notification UI for this, extended with buttons for interaction, instead of a dialog?

For keyboard users we might want to hint at using `:qa!` or `:wqa` for example, which would be a bit awkward if there's a dialog blocking the command line interface.

**Comment by bryphe** [comment]
Sounds reasonable to me - as long as we have a way to interact with the notification with the keyboard (I was thinking `<ESC>` to dismiss and `<CR>/Enter` to continue saving).

Our notifications right now are easily ignored / noisy... but maybe if the non-actionable ones are in the status bar (https://github.com/onivim/oni2/issues/849) and we reserve the current notification UI for cases that require user intervention - it would work.
